codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddTranscribeStreamingDependency.java

```diff
@@ -0,0 +1,81 @@
+package software.amazon.smithy.aws.typescript.codegen;
+
+import static software.amazon.smithy.typescript.codegen.integration.RuntimeClientPlugin.Convention.HAS_CONFIG;
+import static software.amazon.smithy.typescript.codegen.integration.RuntimeClientPlugin.Convention.HAS_MIDDLEWARE;
+
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import software.amazon.smithy.aws.traits.ServiceTrait;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.typescript.codegen.LanguageTarget;
+import software.amazon.smithy.typescript.codegen.TypeScriptDependency;
+import software.amazon.smithy.typescript.codegen.TypeScriptSettings;
+import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+import software.amazon.smithy.typescript.codegen.integration.RuntimeClientPlugin;
+import software.amazon.smithy.typescript.codegen.integration.TypeScriptIntegration;
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.MapUtils;
+
+/**
+ * Add client plugins and configs to support WebSocket streaming for Transcribe
+ * Streaming service
+ * */
+public class AddTranscribeStreamingDependency implements TypeScriptIntegration {
+    @Override
+    public List<RuntimeClientPlugin> getClientPlugins() {
+        return ListUtils.of(
+                RuntimeClientPlugin.builder()
+                        .withConventions(AwsDependency.TRANSCRIBE_STREAMING_MIDDLEWARE.dependency,
+                                "WebSocket")
+                        .servicePredicate((m, s) -> AddTranscribeStreamingDependency.isTranscribeStreaming(s))
+                        .build()
+        );
+    }
+
+    @Override
+    public Map<String, Consumer<TypeScriptWriter>> getRuntimeConfigWriters(
+            TypeScriptSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            LanguageTarget target
+    ) {
+        ServiceShape service = settings.getService(model);
+        if (!isTranscribeStreaming(service)) {
+            return Collections.emptyMap();
+        }
+
+        Map<String, Consumer<TypeScriptWriter>> transcribeStreamingHandlerConfig = MapUtils.of(
+                "eventStreamPayloadHandlerProvider", writer -> {
+            writer.addDependency(AwsDependency.TRANSCRIBE_STREAMING_MIDDLEWARE);
```

Comment:
Formatting: Can these function bodies be moved to 8 spaces inside the line they originate on and the closing `}` be aligned? This will make these lines a more readable, I'll take a follow to see if we can get Checkstyle to enforce this.

```
        Map<String, Consumer<TypeScriptWriter>> transcribeStreamingHandlerConfig = MapUtils.of(
                "eventStreamPayloadHandlerProvider", writer -> {
                        writer.addDependency(AwsDependency.TRANSCRIBE_STREAMING_MIDDLEWARE);
                        ...
                },
                "requestHandler", writer -> {
                        ...
                });